tests/ha/qnetd.pm

```diff
@@ -0,0 +1,106 @@
+
+    # Check split brain situation
+    if ($expected_status eq 'split-brain') {
+        die "Unexpected output for split-brain situation" unless ($output =~ /Activity blocked/);
+        return;
+    }
+
+    my @regexps = map { $_ . ($num_nodes + 1) } ('Expected votes:\s+', 'Highest expected:\s+', 'Total votes:\s+');
+    push @regexps, 'Quorum:\s+' . $num_nodes;
+
+    push @regexps, 'Flags:\s+Quorate\s+Qdevice' if ($expected_status eq 'started');
+    push @regexps, 'Flags:\s+2Node\s+Quorate'   if ($expected_status eq 'stopped');
+
+    die "Qdevice membership information does not match expected info" if ($expected_status eq 'started' and $output !~ /\s+0\s+1\s+Qdevice/);
+    die "Qdevice membership information shown when stopped"           if ($expected_status eq 'stopped' and $output =~ /\s+0\s+1\s+Qdevice/);
+
+    foreach my $exp (@regexps) { die "Unexpected output. Output does not match [$exp]" unless ($output =~ /$exp/) }
+}
+
+sub run {
+    my $cluster_name  = get_cluster_name;
+    my $qdevice_check = "/etc/corosync/qdevice/check_master.sh";
+
+    if (check_var('QDEVICE_TEST_ROLE', 'qnetd_server')) {
+        zypper_call 'in corosync-qnetd';
+        barrier_wait("QNETD_SERVER_READY_$cluster_name");
+    }
+    else {
+        assert_script_run "curl -f -v " . autoinst_url . "/data/ha/qdevice_check_master.sh -o $qdevice_check";
+    }
+
+    if (is_node(1)) {
+        my $qnet_node_host = choose_node(3);
+        my $qnet_node_ip   = get_ip($qnet_node_host);
+        my $node2_ip       = get_ip(choose_node(2));
+
+        # Add a promotable resource to check if the current node is hosting
+        # master instance of the resource. If so, this cluster partition
+        # is preferred to be given the vote from qnetd.
+        assert_script_run "EDITOR=\"sed -ie '\$ a primitive stateful-1 ocf:pacemaker:Stateful'\" crm configure edit";
+        assert_script_run "EDITOR=\"sed -ie '\$ a clone promotable-1 stateful-1 meta promotable=true'\" crm configure edit";
+        save_state;
+
+        # Qdevice should be started
+        qdevice_status('started');
+
+        # Remove qdevice
+        assert_script_run "crm cluster remove --qdevice -y";
+        # Qdevice should be stopped
+        qdevice_status('stopped');
+
+        # Add qdevice to a running cluster with heuristic check
+        assert_script_run "crm cluster init qdevice --qnetd-hostname=$qnet_node_ip -y --qdevice-heuristics=/etc/corosync/qdevice/check_master.sh --qdevice-heuristics-mode=on";
+        # Qdevice should be started again
+        qdevice_status('started');
+
+        record_info('Split-brain info', 'Split brain test');
+        assert_script_run "iptables -A INPUT -s $node2_ip -j DROP; iptables -A OUTPUT -d $node2_ip -j DROP";
+        sleep $default_timeout;
+    }
+
+    barrier_wait("SPLIT_BRAIN_TEST_$cluster_name");
+
+    record_info('Split-brain check', 'Check if activity is blocked') and
+      qdevice_status('split-brain') if is_node(2);
```

Comment:
Is this not equivalent to

    if (is_node(2)) {
        record_info('Split-brain check', 'Check if activity is blocked');
        qdevice_status('split-brain');
    }

If yes maybe more readable, no? But that's can be merged anyway, not so important :D

Reply:
Yes it's more readable but usually, when I'm doing something like your example, someone told me to do it in one line heh :sweat_smile:

Reply:
> Yes it's more readable but usually, when I'm doing something like your example, someone told me to do it in one line eheh

One line if it's only one command/function call. But with 2, I agree with @ldevulder that looks better as an if block.